youtube_dl/extractor/zdf.py

```diff
@@ -6,207 +6,98 @@
+        jsb_json = self._parse_json(jsb, video_id)
 
-        status_code = doc.find('./status/statuscode')
-        if status_code is not None and status_code.text != 'ok':
-            code = status_code.text
-            if code == 'notVisibleAnymore':
-                message = 'Video %s is not available' % video_id
-            else:
-                message = '%s returned error: %s' % (self.IE_NAME, code)
-            raise ExtractorError(message, expected=True)
+        configuration_url = 'https://www.zdf.de' + jsb_json['config']
+        configuration_json = self._download_json(configuration_url, video_id, note='Downloading player configuration')
+        api_token = configuration_json['apiToken']
 
-        title = doc.find('.//information/title').text
-        description = xpath_text(doc, './/information/detail', 'description')
-        duration = int_or_none(xpath_text(doc, './/details/lengthSec', 'duration'))
-        uploader = xpath_text(doc, './/details/originChannelTitle', 'uploader')
-        uploader_id = xpath_text(doc, './/details/originChannelId', 'uploader id')
-        upload_date = unified_strdate(xpath_text(doc, './/details/airtime', 'upload date'))
-        subtitles = {}
-        captions_url = doc.find('.//caption/url')
-        if captions_url is not None:
-            subtitles['de'] = [{
-                'url': captions_url.text,
-                'ext': 'ttml',
-            }]
+        player_js = self._download_webpage('https://www.zdf.de/ZDFplayer/latest-v2/skins/zdf/zdf-player.js', video_id, note='Downloading player script')
+        player_id = self._search_regex(r'this\.ptmd_player_id="([^"]*)"', player_js, 'player id')
 
-        def xml_to_thumbnails(fnode):
-            thumbnails = []
-            for node in fnode:
-                thumbnail_url = node.text
-                if not thumbnail_url:
-                    continue
-                thumbnail = {
-                    'url': thumbnail_url,
-                }
-                if 'key' in node.attrib:
-                    m = re.match('^([0-9]+)x([0-9]+)$', node.attrib['key'])
-                    if m:
-                        thumbnail['width'] = int(m.group(1))
-                        thumbnail['height'] = int(m.group(2))
-                thumbnails.append(thumbnail)
-            return thumbnails
+        content_json = self._download_json(jsb_json['content'], video_id, headers={'Api-Auth': 'Bearer {}'.format(api_token)}, note='Downloading content description')
+        main_video_content = content_json['mainVideoContent']['http://zdf.de/rels/target']
+        meta_data_url_template = main_video_content['http://zdf.de/rels/streams/ptmd-template']
+        meta_data_url = 'https://api.zdf.de' + meta_data_url_template.replace('{playerId}', player_id)
 
-        thumbnails = xml_to_thumbnails(doc.findall('.//teaserimages/teaserimage'))
+        title = main_video_content['title']
```

Comment:
This doesn't seem to be correct, For https://www.zdf.de/service-und-hilfe/die-neue-zdf-mediathek/zdfmediathek-trailer-100.html it returns "Trailer ZDFmediathek Supermarkt" but on the webpage it says "Die neue ZDFmediathek". And with Nachrichten is worse, because for https://www.zdf.de/nachrichten/heute/videos/heute-sendung-vom-30-10-2016-um-19-Uhr-100.html it gives 161030_h19, but should be "ZDF-heute-Sendung vom 30.10.2016". Therefore I would use `content_json['title']`